bin/custom/concerttours/src/concerttours/facades/impl/DefaultBandFacade.java

```diff
@@ -0,0 +1,66 @@
+package concerttours.facades.impl;
+
+import concerttours.data.BandData;
+import concerttours.data.TourSummaryData;
+import concerttours.facades.BandFacade;
+import concerttours.model.BandModel;
+import concerttours.service.BandService;
+import de.hybris.platform.core.model.product.ProductModel;
+import de.hybris.platform.servicelayer.dto.converter.Converter;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Required;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DefaultBandFacade implements BandFacade {
+
+    private BandService bandService;
+    private Converter<BandModel, BandData> bandConverter;
+    private Converter<ProductModel, TourSummaryData> tourSummaryConverter;
+
+    @Override
+    public BandData getBand(String id) throws IllegalArgumentException {
+        if (id == null) {
+            throw new IllegalArgumentException("Band's id can't be null");
+        }
+        BandModel bandModel = bandService.getBandForCode(id);
+        return getBand(bandModel);
+    }
+
+    @Override
+    public List<BandData> getBands() {
+        return CollectionUtils.emptyIfNull(bandService.getBands())
+                .stream()
+                .map(this::getBand)
+                .collect(Collectors.toList());
+    }
+
+    private BandData getBand(BandModel bandModel) {
+        if (bandModel == null) {
+            return null;
+        }
+        BandData bandData = bandConverter.convert(bandModel);
+        List<TourSummaryData> tourSummaryData = CollectionUtils.emptyIfNull(bandModel.getTours())
```

Comment:
This should be invoked in a separate populator which should be included in the band converter's populators list. So that in this class you'll just have to do **bandConverter.convert(model)** and this will already return a fully populated DTO. The getBand method will be unneeded.